docs/State.md

```diff
@@ -0,0 +1,246 @@
+rng.eval(1250496027); // A different seed: 0.5198217719046602
+
+// Chain together to pass the new seed to the next RNG
+// pair :: RNG a -> RNG (Tuple a a)
+const pair = rng => rng.chain(a => rng.chain(b => State.of(Tuple(a, b))));
+
+pair(rng).eval(42); // Tuple(0.5823236793115024, 0.5198217719046602)
+
+// Map to produce transformed random values from 1 to 6
+// rollDie :: RNG Integer
+const rollDie = rng.map(n => Math.ceil(n * 6));
+
+// rollDice :: RNG (Tuple Integer Integer)
+const rollDice = pair(rollDie);
+rollDice.eval(123); // Tuple(2, 5)
+```
+
+## Reference
+
+### Constructors
+
+#### `State`
+```hs
+:: (s -> Identity (Tuple a s)) -> State s a
+```
+Constructs a `State` instance that represent a pure computation from some state
+to a new state and a result. Note this constructor requires the given function
+to return an `Identity` instance. It is generally recommended to use the static
+properties and methods provided on the `State` object rather than using this
+constructor.
+
+#### `State.T`
+```hs
+:: Monad m => { of :: a -> m a } -> (s -> m (Tuple a s)) -> StateT s m a
+```
+Constructs a `StateT` instance that represent a computation from some state to a
+new state and a result in the context of some other monad. It is generally
+recommended to use the static properties and methods provided on the `State`
+object rather than using this constructor.
+
+### Static properties
+
+#### `State.get`
+```hs
+:: State s s
+```
+A static `State` instance that retrieves the current state.
+
+#### `StateT.get`
+```hs
+:: Monad m => StateT s m s
+```
+A static `StateT` instance that retrieves the current state.
+
+### Static methods
+
+#### `State.gets`
+```hs
+:: (s -> a) -> State s a
+```
+Returns a `State` instance the retrieves the current state transformed by the
+given function.
+
+#### `StateT.gets`
+```hs
+:: Monad m => (s -> a) -> StateT s m a
+```
+Returns a `State` instance the retrieves the current state transformed by the
+given function.
+
+#### `State.put`
+```hs
+:: s -> State s a
+```
+Returns a `State` instance the stores the provided state.
+
+#### `StateT.put`
+```hs
+:: Monad m => s -> StateT s m a
+```
+Returns a `StateT` instance the stores the provided state
+
+#### `State.modify`
+```hs
+:: (s -> s) -> State s a
+```
+Returns a `State` instance the modifies the stored state with the provided
+function.
+
+#### `StateT.modify`
+```hs
+:: Monad m => (s -> s) -> StateT s m a
+```
+Returns a `StateT` instance the modifies the stored state with the provided
+function.
+
+#### `State.of`
+```hs
+:: a -> State s a
+```
+Returns a `State` instance that will evaluate to the provided value.
+
+#### `StateT.of`
+```hs
+:: Monad m => a -> StateT s m a
+```
+Returns a `StateT` instance that will evaluate to the provided value.
+
+#### `StateT.lift`
+```hs
+:: Moand m => m a -> StateT s m a
+```
+Lifts the given monad into a `StateT` instance.
+
+### Instance methods
+
+#### `state.run`
```

Comment:
What do you think of using `State#run` to indicate an instance method rather than `state.run`?